### PR TITLE
Enable retries if pyodbc fails to establish a connection

### PR DESCRIPTION
**Pull request by RobbieClarken**
The common Informix errors we see

```
08004  Attempt to connect to database server (<dbname>) failed. (-908) (SQLDriverConnect)
HY000  Read error occurred during connection attempt. (-27001) (SQLDriverConnect) 
```

both happen when django-informixdb attempts to open a new connection. This PR adds retry logic around opening the connection. The delay between tries uses exponential backoff with jitter described here: https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/.

The retry logic is disabled by default but can be configured with settings described in the readme.

I will test this in our development environments before merging but wanted to get early feedback.

### CODE REVIEW

File: django_informixdb/base.py [hunk 1]
```diff
@@ -253,6 +254,39 @@ def get_new_connection(self, conn_params):
...
+        wait_max = retry_params.get("WAIT_MAX", 1000)
+        multiplier = retry_params.get("WAIT_MULTIPLIER", 25)
+        exp_base = retry_params.get("WAIT_EXP_BASE", 2)
+
+        attempt = 0
+        while True:
+            attempt += 1
+            try:
+                conn = pyodbc.connect(connection_string, autocommit=conn_params["AUTOCOMMIT"])
+            except pyodbc.Error as err:
```
Comment (richard-reece) [inline django_informixdb/base.py]: This seems quite broad - will this result in retries if eg. the connection string is misconfigured?

Comment (RobbieClarken) [reply django_informixdb/base.py]: Good point. I've updated the PR to only retry on error codes `-908` and `-27001`.

File: README.rst [hunk 2]
```diff
@@ -100,6 +108,22 @@ LOCK_MODE_WAIT
...
+        random_between(WAIT_MIN, min(WAIT_MAX, WAIT_MULTIPLIER * WAIT_EXP_BASE ** attempt))
+
+    Defaults (wait times are in milliseconds)::
+
+        MAX_ATTEMPTS: 1
+        WAIT_MIN: 0
+        WAIT_MAX: 1000
+        WAIT_MULTIPLIER: 25
+        WAIT_EXP_BASE: 2
+        ERRORS: ['-908', '-27017']
```
Comment (richard-reece) [inline README.rst]: Really minor but for clarity it might be good to explain what these two are any why they're the default.

Comment (RobbieClarken) [reply README.rst]: Done

**Pull request merged**